enclave_mgmt/compile_models.py

```diff
@@ -1,945 +1,9 @@
-        Key=key_ib_pset_problems)
-    course_contents_stream = s3_client.get_object(
-        Bucket=bucket,
-        Key=key_course_contents)
-
-    quiz_question_data = pd.read_csv(
-        BytesIO(quiz_question_stream["Body"].read())
-    )
-    quiz_question_contents_data = pd.read_csv(
-        BytesIO(quiz_question_contents_stream["Body"].read())
-    )
-    quiz_multichoice_answers_data = pd.read_csv(
-        BytesIO(quiz_multichoice_answers_stream["Body"].read()),
-    )
-    ib_input_instances_data = pd.read_csv(
-        BytesIO(ib_input_instances_stream["Body"].read()),
-    )
-    ib_pset_problems_data = pd.read_csv(
-        BytesIO(ib_pset_problems_stream["Body"].read()),
-    )
-    course_contents_data = pd.read_csv(
-        BytesIO(course_contents_stream["Body"].read()),
-    )
-
-    return {"quiz_questions": quiz_question_data,
-            "quiz_question_contents": quiz_question_contents_data,
-            "quiz_multichoice_answers": quiz_multichoice_answers_data,
-            "ib_input_instances": ib_input_instances_data,
-            "ib_pset_problems": ib_pset_problems_data,
-            "course_contents": course_contents_data
-            }
-
-
-def collect_event_data_dfs(events_bucket, events_key):
-
-    key_content_loads = events_key + "/content_loaded_v1.json"
-    key_ib_pset_problem_attempts = \
-        events_key + "/ib_pset_problem_attempted_v1.json"
-    key_ib_input_submissions = events_key + "/ib_input_submitted_v1.json"
-
-    s3_client = boto3.client("s3")
-
-    content_loads_stream = s3_client.get_object(
-        Bucket=events_bucket,
-        Key=key_content_loads)
-    ib_pset_problem_attempts_stream = s3_client.get_object(
-        Bucket=events_bucket,
-        Key=key_ib_pset_problem_attempts)
-    ib_input_submissions_stream = s3_client.get_object(
-        Bucket=events_bucket,
-        Key=key_ib_input_submissions)
-
-    content_loads_data = pd.DataFrame(
-        json.loads(content_loads_stream["Body"].read())['data']
-    )
-
-    # Normalize union type for pset attempt response
-    ib_pset_problem_attempts_json = json.loads(
-        ib_pset_problem_attempts_stream["Body"].read()
-    )['data']
-    ib_pset_problem_attempts_normalized = []
-    for item in ib_pset_problem_attempts_json:
-        item["response"] = \
-            item["response"]["string"] or item["response"]["array"]
-        ib_pset_problem_attempts_normalized.append(item)
-    ib_pset_problem_attempts_data = pd.DataFrame(
-        ib_pset_problem_attempts_normalized
-    )
-
-    ib_input_submissions_data = pd.DataFrame(
-        json.loads(ib_input_submissions_stream["Body"].read())['data']
-    )
-
-    return {
-        "content_loads": content_loads_data,
-        "ib_pset_problem_attempts": ib_pset_problem_attempts_data,
-        "ib_input_submissions": ib_input_submissions_data
-    }
+from enclave_mgmt.data_collection import (
+    collect_content_dfs, collect_event_data_dfs, collect_moodle_dfs)
```

Comment:
If we move the `compile_models()` function (or potentially renamed `collect_data`) into `data_collection.py` (or potentially renamed `collect_data.py`), these "internal" methods won't need to pollute the main file. It also provides some symmetry:

* Import `collect_data()` from `collect_data.py`
* Import `create_models()` from `create_models.py`
* `compile_models.py` invokes `collect_data()` + `create_models()`

What do you think?